packages/as-transformer/src/transformer.ts

```diff
@@ -4,13 +4,27 @@ import {
   Parser,
   CallExpression,
   IdentifierExpression,
+  FunctionDeclaration,
 } from 'assemblyscript/dist/assemblyscript.js';
 import { File2ByteArray } from './transformers/file2ByteArray.js';
 import { TestTable } from './transformers/testTable.js';
 
 const callTransformers = [File2ByteArray, TestTable];
 
+const protobufTransformerDecorator = 'exportUsingPB';
+
 export class Transformer extends TransformVisitor {
+  visitFunctionDeclaration(node: FunctionDeclaration): FunctionDeclaration {
+    if (utils.hasDecorator(node, protobufTransformerDecorator)) {
```

Comment:
At first, I would have said that it would be on all exported functions. Do we use a decorator to avoid being too restrictive?

Reply:
yes we only want to tranform function having a specific decorator. and we should add massa in this decorator to avoid collision